docs/signature-layout.md

```diff
@@ -0,0 +1,35 @@
+# Signature File Layout
+
+A common file layout for storing and serving signatures provides a consistent way to reference image signatures. Signatures on a filesystem or a web server shall use this common layout. Signatures stored in a REST API are not required to use this common layout.
+
+## Specification
+
+This specification relies on [RFC3986](https://tools.ietf.org/html/rfc3986), focusing on defining a [path component](https://tools.ietf.org/html/rfc3986#section-3.3) to compose a concise URI reference to a signature.
+
+**SCHEME[AUTHORITY]/PATH_PREFIX/IMAGE@MANIFEST_DIGEST/signature-INT**
+
+**Definitions**
+
+* **SCHEME**: URI scheme per [RFC3986](https://tools.ietf.org/html/rfc3986#section-3.1), e.g. **file://** or **https://**
+* **AUTHORITY**: An optional authority reference per [RFC3986](https://tools.ietf.org/html/rfc3986#section-3.2), e.g. **example.com**
+* **PATH_PREFIX**: An arbitrary base path to the image component
+* **IMAGE**: The name of the image per [v2 API](https://docs.docker.com/registry/spec/api/#/overview). This would typically take the form of registry/repository/image but is not required to have exactly three parts. There is no requirement to include a **:PORT** component but it should be included if part of the image reference.
```

Comment:
(WRT registry/repo in the URLs relative to the `sigstore` root:) Philosophically, the `sigstore` mechanism is _not_ a preferred way to distribute signatures for images; the preferred way is to have signatures explicitly supported in registries, as “attachments” to a manifest. `sigstore` is essentially a compatibility/interoperability mechanism to be able to use and distribute signatures along with existing docker/distribution registries which don’t have such a manifest attachment mechanism. Basically, think of `sigstore` as, as far as possible, an invisible implementation detail of a registry protocol (Obviously it is not really invisible because the user needs to configure the URLs.)

With a native registry support, the signature attachments would naturally be controlled by the registry’s authentication and authorization mechanisms, and would be namespaced / private exactly as the images themselves are (or are not). So, if a repository in such a registry did not reveal existence of an image in a specific repo to unauthorized users, similarly it should not reveal existence of a signature in that repo to unauthorized users.

A flat namespace, where all signatures for the same image are shared, regardless of which repository they are referenced from, would be a different model. That does not _really_ affect signature validity (extra unrecognized signatures are ignored), but it does affect privacy (it could reveal the existence of internal testing keys, or, perhaps, the stage of QA an image is currently in), and it makes access control _much_ more complicated. With signatures namespaced per-repo, the sigstore server can be set up so that _only_ the owner of each repo on the docker/distribution registry can modify signatures for that repo’s sigstore; with a flat namespace, everyone would probably have to have the access rights to add or remove signatures to any other image, including images of completely different users. (This permission design issue could be worked around by creating an extra sigstore root for each repo, but that would defeat the point of a flat namespace again, and it would be visible to consumers: with a non-flat namespace, ACME Example can publish a single sigstore root at registry.acme.example.com/sigstore, and the customers will never know how the access rights and internal organization units are set up; a set of flat sigstore roots partitioned so that each department has its own would be visible to customers, who would have to set some or all of them, instead of a single one for the whole company.)

And, again, the sigstore is philosophically an invisible implementation detail simulating real registries with native signature support. Real registries with native signature support do not in any way create an internet-wide flat namespace, at least not without a lot of extra effort, so the sigstores should not create such a flat namespace either.